### PR TITLE
Ssse3 optimize and sanitize address

### PR DESCRIPTION
**Pull request by akallabeth**

### CODE REVIEW

**Comment by akallabeth** [comment]
@freerdp-bot test

File: client/common/file.c [hunk 1]
```diff
@@ -490,19 +509,25 @@ static BOOL freerdp_client_parse_rdp_file_option_ascii(rdpFile* file, char* opti
...
 	char* line;
 	char* type;
 	char* context;
-	char *d1, *d2;
-	char *beg, *end;
-	char *name, *value;
+	char* d1, *d2;
+	char* beg, *end;
+	char* name, *value;
+	char* copy = calloc(1, size + sizeof(BYTE));
```
Comment (mfleisz) [inline client/common/file.c]: Leaks if one of the returns in the while loop is executed

File: client/common/file.c [hunk 2]
```diff
@@ -562,36 +586,42 @@ static BOOL freerdp_client_parse_rdp_file_buffer_ascii(rdpFile* file, const BYTE
...
-	WCHAR* line;
+	const WCHAR* line;
 	WCHAR* type;
 	WCHAR* context;
-	WCHAR *d1, *d2;
-	WCHAR *beg, *end;
-	WCHAR *name, *value;
+	WCHAR* d1, *d2;
+	const WCHAR* name, *value;
+	WCHAR* copy = (WCHAR*)calloc(1, size + sizeof(WCHAR));
```
Comment (mfleisz) [inline client/common/file.c]: Leaks if one of the returns in the while loop is executed

**Comment by akallabeth** [comment]
@mfleisz Ok, fixed.

**Comment by mfleisz** [comment]
@akallabeth Leaks are fixed but freerdp_client_parse_rdp_file_buffer_XXX will now always return TRUE - even on errors.

**Comment by akallabeth** [comment]
@freerdp-bot test and don't nag

**Pull request merged**